test-data/unit/fine-grained-modules.test

```diff
@@ -971,3 +971,56 @@ x = Foo()
 ==
 ==
 main:2: error: Too few arguments for "foo" of "Foo"
+
+[case testAddModuleAfterCache1]
+# cmd: mypy main a.py
+# cmd2: mypy main a.py b.py
+# cmd3: mypy main a.py b.py
+import a
+[file a.py]
+pass
+[file a.py.2]
+import b
+b.foo(0)
+[file b.py.2]
+def foo() -> None: pass
+[file b.py.3]
+def foo(x: int) -> None: pass
+[out]
+==
+a.py:2: error: Too many arguments for "foo"
+==
+
+[case testAddModuleAfterCache2]
+# cmd: mypy main a.py
+# cmd2: mypy main a.py b.py
+# cmd3: mypy main a.py b.py
+# flags: --ignore-missing-imports --follow-imports=skip
+import a
+[file a.py]
+import b
+b.foo(0)
+[file b.py.2]
+def foo() -> None: pass
+[file b.py.3]
+def foo(x: int) -> None: pass
+[out]
+==
+a.py:2: error: Too many arguments for "foo"
+==
+
+[case testAddModuleAfterCache3]
```

Comment:
I'd prefer to have more tests around this. For example, test that errors are correctly generated in all the new modules and existing modified modules, and make sure that we can correctly bind references to things defined in all three classes of modules:

1. Cached but unmodified
2. Cached and modified
3. New modules